### PR TITLE
Store test and test resources in sandbox storage

### LINKED ISSUE #1284: [Sandbox] Store test and test resources on Sandbox storage

### PR DESCRIPTION
**Pull request by nulls**
* supported loading and uploading test to/from sandbox storage
* added routing for frontend to sandbox in local mode

It closes #1284

image

TODO:
* change file name for code example
* we provide a list of test files to save-cli, probably we should provide an empty cli arg to test a discovering too

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt [hunk 1]
```diff
@@ -243,6 +265,67 @@ class SandboxView : AbstractView<SandboxViewProps, SandboxViewState>(false) {
...
+            setState {
+                codeText = newCodeText
+                configText = newConfigText
+                setupShText = newSetupShText
+            }
+        }
+    }
+
+    private suspend fun postTestAsText(
+        type: String,
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt:299]: `SandboxStorageKeyType` instead of string?

Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt:299]: `SandboxStorageKeyType` is not available in frontend. I don't want to move it to common place to be able to extract sandbox in future.
Will split it to two endpoints: `upload-test-as-text` and `upload-test-resources-as-text`

**Review by petertrr — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt [hunk 2]
```diff
@@ -125,8 +133,16 @@ class SandboxView : AbstractView<SandboxViewProps, SandboxViewState>(false) {
...
+                onSelectedFileUpdate = { fileType ->
+                    if (state.firstLoad) {
+                        loadTests()
+                    } else selectedFile?.let {
+                        uploadTests()
+                    }
+                    setState {
+                        selectedFile = fileType
+                        firstLoad = false
+                    }
```
[the remark this answers is not in the document]
Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt:145]: Will try

[the remark this answers is not in the document]
Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/SandboxView.kt]: done

**Pull request merged**